pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
+)
+
+const (
+	jsonID       = "@id"
+	jsonType     = "@type"
+	jsonThread   = "~thread"
+	jsonThreadID = "thid"
+)
+
+// Header helper structure which keeps reusable fields
+// Deprecated: Use DIDCommMsg instead of Header
+// TODO: Remove deprecated Header structure https://github.com/hyperledger/aries-framework-go/issues/1075
+type Header struct {
+	ID     string           `json:"@id"`
+	Thread decorator.Thread `json:"~thread"`
+	Type   string           `json:"@type"`
+	// TODO revisit ~purpose, should be generic map [Issue #1037]
+	Purpose []string `json:"~purpose"`
+}
+
+// DIDCommMsg did comm msg
+type DIDCommMsg map[string]interface{}
```

Comment:
This should probably not be in common/service anymore.

It is an implementation rather than a common interface being consumed by other packages.

Reply:
Give me a hint of where to move it. `pkg/common/didcommmsg` ?